src/prefect/cli/register.py

```diff
@@ -75,14 +81,22 @@ def flow(file, name, project, label):
         --label, -l     TEXT    A label to set on the flow, extending any existing labels.
                                 Multiple labels are supported, eg. `-l label1 -l label2`.
 
+        --skip-if-flow-metadata-unchanged       If toggled, passes a serialized hash of the flow to the
+                                                register function's idempotency key. This allows you to
+                                                avoid bumping the registered flow's version if the flow
+                                                metadata (structure) has not changed.
```

Comment:
I don't think we need to explain how this works behind the scenes, the user will generally only care what the effect is.

```suggestion
        --skip-if-flow-metadata-unchanged       If set, the flow will only be re-registered if its
                                                metadata or structure has changed.
```